mkosi/__init__.py

```diff
@@ -2292,6 +2292,31 @@ def run_shell(args: MkosiArgs, config: MkosiConfig) -> None:
         run(cmdline, stdin=sys.stdin, stdout=sys.stdout, env=os.environ, log=False)
 
 
+def run_ctl_with_name(tool: str, args: MkosiArgs, config: MkosiConfig) -> None:
+    if config.output_format not in (OutputFormat.disk, OutputFormat.directory):
+        die(f"{config.output_format} images cannot be inspected with {tool}")
+
+    if (tool_path := find_binary(tool)) is None:
+        die(f"Failed to find {tool}")
+
+    image_arg_name = "root" if config.output_format == OutputFormat.directory else "image"
+    fname = config.output_dir_or_cwd() / config.output
+    cmdline: list[PathString] = [
+        tool_path,
+        f"--{image_arg_name}={fname}",
+        *args.cmdline
+    ]
```

Comment:
This doesn't need its own variable, please inline it in the run() call

Reply:
It's funny. I was something similar yesterday, but when testing whether the type annotation could be omitted mypy wasn't smart enough to not make this `list[Object]`, but it works when inlining. Weird edge case.

Reply:
behrmann I also had that exact issue yesterday 😂 and when I had some slightly more complicated logic for the cmdline that was all that fixed it. I will inline it now :)